$NetBSD: patch-aa,v 1.4 2010/05/06 09:36:28 sbd Exp $

  - make package build with pkgsrc.
  - DESTDIR fixes
  - work around for "cp -r" behavior inconsistencies.


--- Makefile.orig	1995-12-05 12:26:11.000000000 +0000
+++ Makefile
@@ -13,13 +13,13 @@ DIST = SGMLSpm-1.03ii.tar.gz
 #
 
 # Where is the binary for perl5 located on your system?
-PERL = /usr/bin/perl
+PERL = ${PERL5}
 
 # Where do you want the sgmlspl executable script to be installed?
-BINDIR = /usr/local/bin
+BINDIR = ${PREFIX}/bin
 
 # Where do you put local perl5 libaries?
-PERL5DIR = /usr/local/lib/perl5
+PERL5DIR = ${PREFIX}/${PERL5_SUB_INSTALLVENDORLIB}
 MODULEDIR = ${PERL5DIR}/SGMLS
 
 # Where do you want to put sgmlspl specifications?
@@ -29,61 +29,68 @@ SPECDIR = ${PERL5DIR}
 # do you intend to put it?  'make html' will create two
 # subdirectories, ${HTMLDIR}/SGMLSpm and ${HTMLDIR}/sgmlspl, and place
 # its files there.
-HTMLDIR = /usr/local/lib/www/docs
+HTMLDIR = ${PREFIX}/share/doc/p5-SGMLS
 
 
 #
 # End of user options.
 #
 
-HTML_SOURCES = DOC/HTML/SGMLSpm/ DOC/HTML/sgmlspl/
+HTML_SOURCES = DOC/HTML/SGMLSpm DOC/HTML/sgmlspl
 
-FILES =	${BINDIR}/sgmlspl \
-	${PERL5DIR}/SGMLS.pm \
-	${MODULEDIR}/Output.pm \
-	${MODULEDIR}/Refs.pm \
-	${SPECDIR}/skel.pl
+FILES =	${DESTDIR}${BINDIR}/sgmlspl \
+	${DESTDIR}${PERL5DIR}/SGMLS.pm \
+	${DESTDIR}${MODULEDIR}/Output.pm \
+	${DESTDIR}${MODULEDIR}/Refs.pm \
+	${DESTDIR}${SPECDIR}/skel.pl
 
-all: install docs
+all: docs
 
 install: install_system # install_html
 
-install_system: ${MODULEDIR} ${FILES}
+install_system: ${DESTDIR}${MODULEDIR} ${FILES}
 
 dist: ${DIST}
 
-${BINDIR}/sgmlspl: sgmlspl.pl
-	sed -e 's!/usr/bin/perl!${PERL}!' sgmlspl.pl > ${BINDIR}/sgmlspl
-	chmod a+x,a+r ${BINDIR}/sgmlspl
-
-${PERL5DIR}/SGMLS.pm: SGMLS.pm
-	cp SGMLS.pm ${PERL5DIR}/SGMLS.pm
-	chmod a+r ${PERL5DIR}/SGMLS.pm
-
-${MODULEDIR}:
-	if [ ! -d ${MODULEDIR} ]; then\
-	  mkdir ${MODULEDIR}; \
-	  chmod a+x ${MODULEDIR}; \
+${DESTDIR}${BINDIR}/sgmlspl: sgmlspl.pl
+	sed -e 's!/usr/bin/perl!${PERL}!' sgmlspl.pl > \
+		${DESTDIR}${BINDIR}/sgmlspl
+	chmod a+x,a+r ${DESTDIR}${BINDIR}/sgmlspl
+
+${DESTDIR}${PERL5DIR}/SGMLS.pm: SGMLS.pm
+	cp SGMLS.pm ${DESTDIR}${PERL5DIR}/SGMLS.pm
+	chmod a+r ${DESTDIR}${PERL5DIR}/SGMLS.pm
+
+${DESTDIR}${MODULEDIR}:
+	if [ ! -d ${DESTDIR}${MODULEDIR} ]; then\
+	  mkdir ${DESTDIR}${MODULEDIR}; \
+	  chmod a+x ${DESTDIR}${MODULEDIR}; \
 	fi
 
-${MODULEDIR}/Output.pm: Output.pm
-	cp Output.pm ${MODULEDIR}/Output.pm
-	chmod a+r ${MODULEDIR}/Output.pm
-
-${MODULEDIR}/Refs.pm: Refs.pm
-	cp Refs.pm ${MODULEDIR}/Refs.pm
-	chmod a+r ${MODULEDIR}/Refs.pm
-
-${SPECDIR}/skel.pl: skel.pl
-	cp skel.pl ${SPECDIR}/skel.pl
-	chmod a+r ${SPECDIR}/skel.pl
+${DESTDIR}${MODULEDIR}/Output.pm: Output.pm
+	cp Output.pm ${DESTDIR}${MODULEDIR}/Output.pm
+	chmod a+r ${DESTDIR}${MODULEDIR}/Output.pm
+
+${DESTDIR}${MODULEDIR}/Refs.pm: Refs.pm
+	cp Refs.pm ${DESTDIR}${MODULEDIR}/Refs.pm
+	chmod a+r ${DESTDIR}${MODULEDIR}/Refs.pm
+
+${DESTDIR}${SPECDIR}/skel.pl: skel.pl
+	cp skel.pl ${DESTDIR}${SPECDIR}/skel.pl
+	chmod a+r ${DESTDIR}${SPECDIR}/skel.pl
 
 install_html: ${HTML_SOURCES}
 	cd DOC; make html
-	rm -rf ${HTMLDIR}/SGMLSpm ${HTMLDIR}/sgmlspl
-	cp -r ${HTML_SOURCES} ${HTMLDIR}
-	chmod a+x,a+r ${HTMLDIR}/SGMLSpm ${HTMLDIR}/sgmlspl
-	chmod a+r ${HTMLDIR}/SGMLSpm/* ${HTMLDIR}/sgmlspl/*
+	if [ ! -d ${DESTDIR}${HTMLDIR} ]; then\
+	  mkdir ${DESTDIR}${HTMLDIR}; \
+	  chmod a+x ${DESTDIR}${HTMLDIR}; \
+	fi
+	rm -rf ${DESTDIR}${HTMLDIR}/SGMLSpm ${DESTDIR}${HTMLDIR}/sgmlspl
+	cp -r ${HTML_SOURCES} ${DESTDIR}${HTMLDIR}
+	rm -f ${DESTDIR}${HTMLDIR}/SGMLSpm/sample.pl
+	cp DOC/sample.pl ${DESTDIR}${HTMLDIR}/SGMLSpm
+	chmod a+x,a+r ${DESTDIR}${HTMLDIR}/SGMLSpm ${DESTDIR}${HTMLDIR}/sgmlspl
+	chmod a+r ${DESTDIR}${HTMLDIR}/SGMLSpm/* ${DESTDIR}${HTMLDIR}/sgmlspl/*
 
 docs:
 	cd DOC; make all
